### PR TITLE
add mercator resources.json + fixtures

### PR DESCRIPTION
**Pull request by der-john**
Prevents #2393.

And adds fixtures for Mercator 2015 + 2016.

### CODE REVIEW

**Comment by joka** [comment]
we recently added the bin/sd_fixture script to fix this kind of issues.
Can you add test_fixture für mercator? (for example see  adhocracy_core.test_fixture/ and adhocracy_core.__init__.includeme) Then we can also remove src/mercator/mercator/tests/acceptance/setup_tests.sh .

**Comment by der-john** [comment]
When I run
> bin/ad_fixtures etc/develeopment.ini
my new fixtures aren't shown as registered. How do I register them?

**Comment by 2e2a** [comment]
> my new fixtures aren't shown as registered. How do I register them?

The folder for fixtures is `fixture/`

**Comment by der-john** [comment]
Thanks. I changed the name, but they're still not registered.

**Comment by 2e2a** [comment]
> Thanks. I changed the name, but they're still not registered.

Should be fixed with last commit

**Comment by joka** [comment]
We should update the documentation regarding how to register a fixture.
You should add to the includeme function of the wantend backend customization package:
from adhocracy_core.interfaces import IFixtureAsset
 `config.registry.registerUtility('' ", IFixtureAsset, name='adhocracy_mercator:fixture')`

The important part ist the `name` keyword. This is a `pkg_resource`definition based of <python_package>:<relative path> or a just <absolute_file_system_path>

**Comment by joka** [comment]
works now :)

**Comment by joka** [comment]
The name of the fixture directory can be arbitrary, but should descript the fixtures, like 'test_fixture'

**Comment by joka** [comment]
LGTM

**Comment by joka** [comment]
Note: We could also  simplify registration with a config method like:

`config.register_fixture_directory(''my_fixtures_directory")

File: src/adhocracy_mercator/adhocracy_mercator/__init__.py [hunk 1]
```diff
@@ -18,6 +19,8 @@ def includeme(config):
     config.include('.resources')
     config.include('.evolution')
     config.add_translation_dirs('adhocracy_core:locale/')
+    config.registry.registerUtility('', IFixtureAsset,
+                                    name='adhocracy_euth:fixture')
```
Comment (joka) [inline src/adhocracy_mercator/adhocracy_mercator/__init__.py]: you register the fixtures from the euth package here.

**Comment by xi** [comment]
I dont't see the value of this pull request: It simply duplicates `src/adhocracy_mercator/adhocracy_mercator/workflows/test_mercator/resources.json` and `src/adhocracy_mercator/adhocracy_mercator/workflows/test_mercator2/resources.json`.

IMHO this PR makes the situation even more confusing than it already is. Instead, we should switch to the new fixture system completely (for all projects and removing the old files).

**Comment by 2e2a** [comment]
The resource files in `/workflows/test_some_process` are there for testing of the process workflows and we just misused them for import.

I agree that adding `src/adhocracy_mercator/adhocracy_mercator/workflows/test_mercator/resources.json` does not make sense, as it is not used in the tests.

But in general duplication can occur if the same data is used to create local processes and for the workflow tests.

**Comment by joka** [comment]
> The resource files in /workflows/test_some_process are there for testing of the process workflows and we just misused them for import.
> 
> I agree that adding src/adhocracy_mercator/adhocracy_mercator/workflows/test_mercator/resources.json does not make sense, as it is not used in the tests.

yes this file should be removed

> But in general duplication can occur if the same data is used to create local processes and for the workflow tests.

We could also use the test_fixture for functional tests, that was one intended use case.
But this does not need to be part of this pull request.

**Comment by der-john** [comment]
> we should switch to the new fixture system completely (for all projects and removing the old files).

I'll do that some time soon. (It's low prio, though.)

**Comment by xi** [comment]
I cherry-picked 278a153 as c3d2e6fb88ef01f4bc8bb09ba1352a98cb488eea because it already adds value. Please open a new pull request when you convert the fixtures to the new system.

**Pull request closed without merging**